src/modules/interfaces.ts

```diff
@@ -479,13 +516,74 @@ export class ModuleBase {
     afterInstallSqlPath?: string;
     beforeUninstallSqlPath?: string;
   };
+  rpc?: { [key: string]: RpcInterface };
   migrations: {
     install: (q: QueryRunner) => Promise<void>;
     afterInstall?: (q: QueryRunner) => Promise<void>;
     beforeRemove?: (q: QueryRunner) => Promise<void>;
     remove: (q: QueryRunner) => Promise<void>;
   };
 
+  private getRpcSql(): [string, string] {
+    let afterInstallSql = '';
+    let beforeUninstallSql = '';
+    for (const [key, rpc] of Object.entries(this.rpc ?? {})) {
+      const rpcOutputEntries = Object.entries(rpc.output ?? {});
+      const rpcOutputTable = rpcOutputEntries
+        .map(([columnName, columnType]) => `${columnName} ${columnType}`)
+        .join(', ');
+      afterInstallSql += `
+        create or replace function ${rpc.name}(variadic _args text[] default array[]::text[]) returns table (
```

Comment:
oh, this would be nice, could be a follow-up improvement, though.

> Does Postgres complain if you did something like `SELECT * FROM my_rpc_fn(3);` or will it auto-cast to text for you?
Yes, it complains it does not exist any function with that name waiting for an `integer` parameter.